e2e-cypress/files/support/e2e.ts

```diff
@@ -0,0 +1,69 @@
+//For more details, you can follow this link:
+//https://docs.cypress.io/guides/end-to-end-testing/azure-active-directory-authentication#Microsoft-AAD-Application-Setup
+function loginViaAAD(username: string, password: string) {
+
+  //Go to your application URL and trigger the login.
+  cy.visit(Cypress.env('your_app_url'))
+
+  //If needed, navigate and click on the login button. 
+  //As an example:
+  //cy.get('button#signIn').click()
+
+  //Login to your AAD tenant.
+  //ATENTION: The redirection can happen at the 'login.microsoftonline.com' and also it might redirect as well to 'login.live.com'
+  cy.origin(
+    'https://login.microsoftonline.com',
+    {
+      args: {
+        username,
+        password,
+      },
+    },
+    ({ username, password }) => {
+      cy.get('input[type="email"]').type(username, {
+        log: false,
+      })
+      cy.get('input[type="submit"]').click()
+      cy.get('input[type="password"]').type(password, {
+        log: false,
+      })
+      cy.get('input[type="submit"]').click()
+    }
+  )
+
+  //Depending on the user and how they are registered with Microsoft, the origin may go to live.com
```

Comment:
Can this be somehow detected on the run so that the commented out code can be used depending on the situation?

Reply:
This is an example of Login, and each project should adapt it to their needs, so I don't really see a need of providing something complex here.